contracts/0.4.24/oracle/LidoOracle.sol

```diff
@@ -328,6 +338,25 @@ contract LidoOracle is ILidoOracle, AragonApp {
         return (minReportableEpochId, getCurrentEpochId());
     }
 
+    /**
+     * @notice Reports beacon balance and its change
+     */
+    function getLastCompletedReports()
+        public view
+        returns (
+            uint256 lastTotalPooledEther,
+            uint256 prevTotalPooledEther,
+            uint256 timeElapsed
+        )
+    {
+        BeaconSpec memory beaconSpec = _getBeaconSpec();
+        CompletedReport storage last = lastCompletedReports[COMPLETED_REPORT_ID_LAST];
+        CompletedReport storage prev = lastCompletedReports[COMPLETED_REPORT_ID_PREV];
+        lastTotalPooledEther = last.totalPooledEther;
+        prevTotalPooledEther = prev.totalPooledEther;
+        timeElapsed = (last.epochId - prev.epochId).mul(beaconSpec.slotsPerEpoch).mul(beaconSpec.secondsPerSlot);
```

Comment:
I'm pretty sure we'd be better off caching this and balance before report instead of having two reports stored

Reply:
Right, implemented this.